libs/transactions/transfer.ts

```diff
@@ -0,0 +1,84 @@
+import type Web3 from "web3";
+import path from "path";
+import { parentLogger } from "../utils";
+import { submitTransaction } from "./helpers/submit";
+import type { Client } from "../client";
+import type { NightfallZkpKeys } from "../nightfall/types";
+import type { TransactionReceipt } from "web3-core";
+import { Transaction } from "libs/types";
+
+const logger = parentLogger.child({
+  name: path.relative(process.cwd(), __filename),
+});
+
+/**
+ * @method handleTransfer handle the flow for transferences in layer2
+ * @param {string} ercAddress - the address of the smart contract created for this token
+ * @param {string} ercStandard - the erc standard
+ * @param {string} ownerAddress - the address of who is doing the transfer
+ * @param {string} ownerPrivateKey - the private key of the sender to sing the transaction
+ * @param {NightfallZkpKeys} ownerZkpKeys - the zkp keys for the sender
+ * @param {string} shieldContractAddress - the address of the Shield smart contract
+ * @param {string} value - the value to be transfered
+ * @param {string} fee - the amount (Wei) to pay a proposer for the transaction
+ * is being taken.  Note that the Nightfall_3 State.sol contract must be approved
+ * by the token's owner to be able to withdraw the token
+ * @param {Web3} web3 - web3js instance
+ * @param {Client} client -
+ * @param {string} recipientAddress - the zkp public key of the recipient
+ * @param {boolean} offchain - indicates if the transfer is onchain or offchain
+ * @returns
+ */
+export async function handleTransfer(
+  ercAddress: string,
+  ercStandard: string,
+  ownerAddress: string,
+  ownerPrivateKey: string,
+  ownerZkpKeys: NightfallZkpKeys,
+  shieldContractAddress: string,
+  value: string,
+  fee: string,
+  web3: Web3,
+  client: Client,
+  recipientAddress: string,
+  offchain: boolean,
+): Promise<{ txL1: TransactionReceipt; txL2: Transaction }> | null {
+  logger.debug("createAndSubmitTransfer");
```

Comment:
:eyes: This log was meant for the name of the function, yours is `handleTransfer`?! But again, I think we should have them all the same. If this was an internal app it wouldn't matter much. But because other developers might need it, the more consistency the better. Also for new people joining.

Reply:
In this case I don't see createAndSubimit... name correct for this function. Because if you see, the function that is doing this (creating a TX and broadcasting this) is the subimitTransaction, which IMO should be called createAndSubmitTransaction.